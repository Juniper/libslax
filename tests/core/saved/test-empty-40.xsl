<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" xmlns:slax="http://xml.libslax.org/slax" version="1.0" extension-element-prefixes="slax-ext slax">
  <xsl:variable name="mine-temp-1">
    <a>alpha</a>
    <b>bravo</b>
    <c>charlie</c>
    <d>delta</d>
  </xsl:variable>
  <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="mine" select="slax-ext:node-set($mine-temp-1)"/>
  <xsl:template match="/">
    <top>
      <one>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:join(&quot;:&quot;, $mine)"/>
      </one>
      <two>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:join(&quot;%20&quot;, &quot;this&quot;, &quot;is&quot;, &quot;the&quot;, &quot;end&quot;, $mine/d)"/>
      </two>
      <three>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:join(/nothing, $mine)"/>
      </three>
      <four>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="concat(&quot;[&quot;, slax:join(&quot;] [&quot;, $mine), &quot;]&quot;)"/>
      </four>
      <xsl:variable name="f">
        <xsl:call-template name="answer"/>
      </xsl:variable>
      <five>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="concat(&quot;fish: &quot;, slax:join(&quot;, &quot;, $f))"/>
      </five>
    </top>
  </xsl:template>
  <xsl:template name="answer">
    <fish>one</fish>
    <fish>two</fish>
    <fish>red</fish>
    <fish>blue</fish>
  </xsl:template>
</xsl:stylesheet>
