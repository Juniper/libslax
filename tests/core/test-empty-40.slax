version 1.3;

var $mine := {
    <a> "alpha";
    <b> "bravo";
    <c> "charlie";
    <d> "delta";
}

main <top> {
    <one> slax:join(":", $mine);
    <two> slax:join("%20", "this", "is", "the", "end", $mine/d);
    <three> slax:join(/nothing, $mine);
    <four> "[" _ slax:join("] [", $mine) _ "]";
    var $f = call answer;
    <five> "fish: " _ slax:join(", ", $f);
}

template answer {
    <fish> "one";
    <fish> "two";
    <fish> "red";
    <fish> "blue";
}

    
