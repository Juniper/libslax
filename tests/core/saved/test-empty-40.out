<?xml version="1.0"?>
<top>
  <one>alpha:bravo:charlie:delta</one>
  <two>this%20is%20the%20end%20delta</two>
  <three>alphabravocharliedelta</three>
  <four>[alpha] [bravo] [charlie] [delta]</four>
  <five>fish: one, two, red, blue</five>
</top>
